service/storage_proxy.cc

```diff
@@ -1726,11 +1726,17 @@ class storage_proxy::cancellable_write_handlers_list : public bi::list<abstract_
         // handler is being removed from the b::list, so if any live iterator points at it,
         // move it to the next object (this requires that the list is traversed in the forward
         // direction).
+        bool drop_end = false;
         for (auto& itp : _live_iterators) {
             if (&**itp == handler) {
                 ++*itp;
+                drop_end |= (*itp == end());
             }
         }
+        if (drop_end) {
+            auto it = std::remove_if(_live_iterators.begin(), _live_iterators.end(), [this] (iterator* pit) { return *pit == end(); });
+            _live_iterators.erase(it, _live_iterators.end());
```

Comment:
nit, how about
```c++
if (drop_end) {
    const auto [first, last] = std::ranges::remove_if(_live_iterators, [this] (iterator* pit) { return *pit == end(); });
    _live_iterators.erase(first, last);
}
```

simpler this way.

Reply:
OK. Pushed. I forget that we can use ranges now.

Reply:
s/can/should/